funnel/templates/event.html

```diff
@@ -46,37 +46,251 @@
+          }
+          else {
+            window.Queue.dequeue("cancelcheckin-queue", data.participant_id);
+          }
+        }
+      });
+    }
+
+    function checkQueues() {
+      //Read first participant from 'checkin-queue' and post check-in to server
+      var participantID = window.Queue.peek("checkin-queue");
+      if(participantID) {
+        postCheckinStatus(participantID, true);
+      }
+      //Read first participant from 'cancelcheckin-queue' and post cancel check-in to server
+      participantID = window.Queue.peek("cancelcheckin-queue");
+      if(participantID) {
+        postCheckinStatus(participantID, false);
+      }
+    }
+
+    //Get the list of participants from server
+    getParticipantsData();
+
+    //If localStorage is available, register event handlers for check-in, cancel-checkin and abort
+    if (Modernizr.localstorage) {
+      $('#participants-table-content').on('click', '.js-checkin', function(event) {
+        event.preventDefault();
+        //Add to participant_id to "checkin-queue"
+        window.Queue.enqueue("checkin-queue", $(this).data('pid'));
+        $(this).siblings('.loader').show();
+      });
+
+      $('#participants-table-content').on('click', '.js-cancel-checkin', function(event) {
+        event.preventDefault();
+        //Add to participant_id to "cancelcheckin-queue"
+        window.Queue.enqueue("cancelcheckin-queue", $(this).data('pid'));
+        $(this).siblings('.loader').show();
+      });
+
+      $('#participants-table-content').on('click', '.js-abort', function(event) {
+        event.preventDefault();
+        var participantID = $(this).siblings('a').data('pid');
+        var participant = "[data-pid=" + participantID +"]";
+
+        if($(participant).hasClass('js-checkin')) {
+          // On abort, if participantID present in the queue, then remove from "checkin-queue"
+          // On abort, if participantID is not present in the "checkin-queue" it implies check-in has already been done, so a cancel check-in requires to be send(add it to "cancelcheckin-queue")
+          if(!window.Queue.dequeue("checkin-queue", participantID)) {
+            window.Queue.enqueue("cancelcheckin-queue", participantID);
+          }
+        }
+        else{
+          if(!window.Queue.dequeue("cancelcheckin-queue", participantID)) {
+            window.Queue.enqueue("checkin-queue", participantID);
+          }
+        }
+        $(participant).siblings('.loader').hide();
+      });
+
+      //If there is no network, display a confirmation dialog when user closes/reloades the page
+      window.onbeforeunload = function() {
+        if(!navigator.onLine) {
+          return 'There is no network!';
+        }
+      }
+
+      //Read 'checkin-queue' and 'cancelcheckin-queue' every 8 second
+      setInterval(checkQueues, 8000);
+
+      //Get participants data from server every 1 min
+      setInterval(getParticipantsData, 60000);
```

Comment:
It's better to co-locate this with the first time the call to `getParticipantsData` is made.